service/src/main/kotlin/fi/espoo/evaka/shared/auth/Acl.kt

```diff
@@ -212,6 +213,23 @@ WHERE employee_id = :userId AND p.id = :pairingId
         }
     )
 
+    fun getRolesForPedagogicalDocument(user: AuthenticatedUser, childId: ChildId): AclAppliedRoles = AclAppliedRoles(
+        (user.roles - UserRole.SCOPED_ROLES) + Database(jdbi).read {
+            it.createQuery(
+                // language=SQL
+                """
+SELECT role
+FROM pedagogical_document pd
+JOIN child_acl_view ON pd.child_id = child_acl_view.child_id
+WHERE employee_id = :userId AND pd.child_id = :childId
```

Comment:
Should validate this once more. The table pedagogical_document has no real use here.